pkg/cache/external/scheduler_cache.go

```diff
@@ -132,6 +132,45 @@ func (cache *SchedulerCache) removeNode(node *v1.Node) error {
 	return nil
 }
 
+func (cache *SchedulerCache) AddPendingPodAllocation(podKey string, nodeID string) {
+	cache.lock.Lock()
+	defer cache.lock.Unlock()
+	delete(cache.inProgressAllocations, podKey)
```

Comment:
Is this delete necessary? Under what circumstances can we have an entry?

As I can see it, `inProgressAllocations` is populated from `StartPodAllocation()` but that already requires that pending has an entry which we just add in line 139. Am I missing something here?

Is it a bug if we already have something in `inProgressAllocations` at this point for the same `podKey`?

Reply:
If a pod fails to bind in the default scheduler for some reason, the only notice we may get (as there is not a callback in the scheduler framework for failed binds) is another call to PreFilter() / Filter() for this pod. So it is possible in that case for us to have an in-progress allocation. So, we remove it and reset the state back to pending to be safe.